subresource "hosting environment" has space in it and is valid

### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.7.5

### AzureRM Provider Version

3.96.0

### Affected Resource(s)/Data Source(s)

azurerm_private_endpoint

### Terraform Configuration Files

```hcl
azurerm_private_endpoint does not take spaces in subresource. However, "hosting environment" is a valid subresource type and needs to be supported.
```


### Debug Output/Panic Output

```shell
╷
│ Error: private_service_connection.0.subresource_names.0 must begin and end with a alphanumeric character, be between 3 and 63 characters in length, only contain letters, numbers, underscores, periods, and dashes
│ 
│   with azurerm_private_endpoint.example,
│   on lambda.tf line 91, in resource "azurerm_private_endpoint" "example":
│   91:     subresource_names              = ["hosting environment","sites"]
│ 
╵
Releasing state lock. This may take a few moments...
Error: Process completed with exit code 1.
```


### Expected Behaviour

We should support this special subresource type as per this documentation: https://learn.microsoft.com/en-us/azure/private-link/private-endpoint-overview#private-link-resource

### Actual Behaviour

Throws an error and validation fails for any spaces in the subresource string.

### Steps to Reproduce

_No response_

### Important Factoids

_No response_

### References

_No response_

--- a/internal/services/network/validate/private_link_subresource_name.go
+++ b/internal/services/network/validate/private_link_subresource_name.go
@@ -11,9 +11,9 @@ import (
 
 func PrivateLinkSubResourceName(i interface{}, k string) (_ []string, errors []error) {
 	// empty is valid; otherwise the name must begin and end with an alphanumeric character, be between
-	// 3 and 63 characters in length and only contain letters, numbers, underscores, periods, and dashes
+	// 3 and 63 characters in length and only contain letters, numbers, underscores, periods, dashes, and spaces
 	return validation.Any(
 		validation.StringIsEmpty,
-		validation.StringMatch(regexp.MustCompile(`^([a-zA-Z0-9])([\w\.-]{1,61})([a-zA-Z0-9])$`), "must begin and end with a alphanumeric character, be between 3 and 63 characters in length, only contain letters, numbers, underscores, periods, and dashes"),
+		validation.StringMatch(regexp.MustCompile(`^([a-zA-Z0-9])([\w .-]{1,61})([a-zA-Z0-9])$`), "must begin and end with an alphanumeric character, be between 3 and 63 characters in length, only contain letters, numbers, underscores, periods, dashes, and spaces"),
 	)(i, k)
 }

--- a/internal/services/network/validate/private_link_subresource_name.go
+++ b/internal/services/network/validate/private_link_subresource_name.go
@@ -17,8 +17,8 @@ func PrivateLinkSubResourceName(i interface{}, k string) (_ []string, errors []e
 	}
 
 	if len(strings.TrimSpace(v)) >= 3 {
-		if m, _ := validate.RegExHelper(i, k, `^([a-zA-Z0-9])([\w\.-]{1,61})([a-zA-Z0-9])$`); !m {
-			errors = append(errors, fmt.Errorf("%s must begin and end with a alphanumeric character, be between 3 and 63 characters in length, only contain letters, numbers, underscores, periods, and dashes", k))
+		if m, _ := validate.RegExHelper(i, k, `^([a-zA-Z0-9])([\w .-]{1,61})([a-zA-Z0-9])$`); !m {
+			errors = append(errors, fmt.Errorf("%s must begin and end with a alphanumeric character, be between 3 and 63 characters in length, only contain letters, numbers, underscores, periods, dashes, and spaces", k))
 		}
 	} else if len(v) != 0 {
 		errors = append(errors, fmt.Errorf("%s must be at least 3 character in length", k))

--- a/internal/services/network/validate/private_link_subresource_name_test.go
+++ b/internal/services/network/validate/private_link_subresource_name_test.go
@@ -51,6 +51,31 @@ func TestPrivateLinkSubResourceName(t *testing.T) {
 		{
 			Name:  "Blob Space Secondary",
 			Input: "blob secondary",
+			Valid: true,
+		},
+		{
+			Name:  "ADME Sub-Resource With Spaces",
+			Input: "Azure Data Manager for Energy",
+			Valid: true,
+		},
+		{
+			Name:  "App Service Environment Sub-Resource With Space",
+			Input: "hosting environment",
+			Valid: true,
+		},
+		{
+			Name:  "Power BI Sub-Resource With Space",
+			Input: "Power BI",
+			Valid: true,
+		},
+		{
+			Name:  "Start With Space",
+			Input: " SqlServer",
+			Valid: false,
+		},
+		{
+			Name:  "End With Space",
+			Input: "SqlServer ",
 			Valid: false,
 		},
 		{

